added: initial draft

HaramChoi-askui opened the pull request:
Initial draft for the flutter app automation example

HaramChoi-askui committed:
Added: Readme.md, demo_app, askui-test

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,368 @@
+cd flutter-example-automation;
+flutter create demo_app;
+```
+2. Install dependencies for the Flutter demo app:
````
The numeration is off here: Should be 3.
do not forget the following numerations ;)

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,368 @@
+```bash
+git clone https://github.com/askui/flutter-example-automation;
+cd flutter-example-automation;
+flutter create demo_app;
````
Maybe add a `cd demo_app`here. I stumbled upon it as it only says to change into that directory in the text and not in code.

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,368 @@
+3. Find your device:
+```bash
+# make sure that your android device is connected, and the USB debugging mode is enabled.
+adb devices;
````
shell command does not need a _;_ at the end

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,368 @@
+4. Install the ADBkeyboard on the device:
+```bash
+# inside ADBKeyBoard-2.0/
+adb -s <your device id> install ADBKeyboard.apk;
````
shell command does not need a _;_ at the end

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,368 @@
+5. Configure the ADB Keyboard:
+```bash
+# inside ADBKeyBoard-2.0/
+adb -s <your device id> shell settings put secure default_input_method com.android.adbkeyboard/.AdbIME;
````
shell command does not need a _;_ at the end

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,368 @@
+
+### 3. Setup askui
+1. Setup askui by following [these steps](https://docs.askui.com/docs/general/Getting%20Started/getting-started).
+2. We need to run the uiController directly with an extra argument to specify the runtime mode, as the current version of askui(ver. 0.3.2) doesn't provide the api for running it with the runtime argument yet. <br><br>From within your npm project path, go to the directory that contains the askui-ui-controller binary, and run `./askui-ui-controller -r android` with an optional argument for the and ...
```
Maybe remove the last sentence _with an optional argument for the android runtime_. For me, it was clear from the command that we are starting the UiController with the Android flag.

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,368 @@
+2. We need to run the uiController directly with an extra argument to specify the runtime mode, as the current version of askui(ver. 0.3.2) doesn't provide the api for running it with the runtime argument yet. <br><br>From within your npm project path, go to the directory that contains the askui-ui-controller binary, and run `./askui-ui-controller -r android` with an optional argument for the and ...
+```bash
+cd <YOUR_PROJECT_DIRECTORY>/node_modules/askui/dist/release/latest/<YOUR_PLATFORM>
+./askui-ui-controller -r android;
````
shell command does not need a ; at the end

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,368 @@
+
+# for example, Mac OS
+cd node_modules/askui/dist/release/latest/darwin/askui-ui-controller.app/Contents/MacOS/
+./askui-ui-controller -r android;
```
shell command does not need a ; at the end

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,368 @@
+cd node_modules/askui/dist/release/latest/darwin/askui-ui-controller.app/Contents/MacOS/
+./askui-ui-controller -r android;
+```
+*If you can't find the binary as described above, then you might have askui freshly installed and haven't run it yet. The binary gets downloaded as the askui test code runs. [See here](https://docs.askui.com/docs/general/Getting%20Started/writing-your-first-test/#run)*
````
Maybe add the command to run the first test here so the user does not have to switch to the docs: `npx jest test/my-first-askui-test-suite.test.ts --config ./test/jest.config.ts`

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,368 @@
+        await aui.type('askui@askui.com').exec();
+        // Scroll down the page
+        // execOnShell() can run shell commands within the device via adb
+        await aui.execOnShell('input swipe 300 600 300 300').exec();
```
I stumbled upon the swipe when testing although it was described in the text earlier. Can we add a brief instruction on what to try if it does not work?
For example: scroll more in x range -> change it to ...

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,368 @@
+
+        // Let's go to the next test page by using horisontal swipe!
+        // Here we swipe two times in a row
+        await aui.execOnShell('input swipe 1500 600 300 600').exec();
```
I stumbled upon the swipe when testing although it was described in the text earlier. Can we add a brief instruction on what to try if it does not work?
For example: scroll more in x range -> change it to ...

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,368 @@
+        // Let's go to the next test page by using horisontal swipe!
+        // Here we swipe two times in a row
+        await aui.execOnShell('input swipe 1500 600 300 600').exec();
+        await aui.execOnShell('input swipe 1500 600 300 600').exec();
```
I stumbled upon the swipe when testing although it was described in the text earlier. Can we add a brief instruction on what to try if it does not work?
For example: scroll more in x range -> change it to ...

JohannesDienst-askui commented on askui-test/demo-automation.ts:
```diff
@@ -0,0 +1,80 @@
+        await aui.click().text().withText('Refuse').exec();
+        await aui.pressAndroidKey('enter').exec();
+        await aui.execOnShell('input swipe 300 600 300 100').exec();
+        
```
Remove excessive newlines and spaces

JohannesDienst-askui commented on askui-test/demo-automation.ts:
```diff
@@ -0,0 +1,80 @@
+        await aui.click().button().contains().text().withText('Submit').exec();
+        await aui.click().text().withText('Refuse').exec();
+        await aui.pressAndroidKey('enter').exec();
+        await aui.execOnShell('input swipe 300 600 300 100').exec();
```
I stumbled upon the swipe when testing although it was described in the text earlier. Can we add a brief instruction on what to try if it does not work?
For example: scroll more in x range -> change it to ...

JohannesDienst-askui commented on askui-test/demo-automation.ts:
```diff
@@ -0,0 +1,80 @@
+    });
+
+
+    it('should pick the dates', async () => {
```
Datepicker-Test  does not work if the whole form is visible (Like with a tablet in landscape format): A hint, that the swipe commands should be removed then is helpful

JohannesDienst-askui commented on askui-test/demo-automation.ts:
```diff
@@ -0,0 +1,80 @@
+        await aui.click().text().withText('edit').nearestTo().text().withText('Depature').exec();
+        await aui.click().icon().withText('chevron right').exec();
+        await aui.click().icon().withText('chevron right').exec();
+        await aui.click().text().withText('20').exec();
```
Selecting 20 can be challenging as askui recognizes it sometimes as 2C.

JohannesDienst-askui reviewed (changes requested):
Nice work!

Some minor things only.

Also do not forget to change the code in README.md and in askui/demo-automation.ts ;-)

HaramChoi-askui committed:
fixed: README.md, demo-automation.ts

1. Fixes based on the change request
2. Newly branched, as the binary part may have to change soon.

HaramChoi-askui committed:
fixed: README.md

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,422 @@
-# flutter-example-automation
-Basic Flutter app with askui automation examples
+# Askui automation example with Flutter
+This repository contains examples of using [askui](https://www.askui.com/) to automate an android app built with [Flutter](https://flutter.dev/). To run the example, setup the provided demo app by following the instruction. This example assumes that you already have the [Android Emulator](https://developer.android.com/studio/run/emulator) installed.
```
Sorry for the nitpicking here:
 
 setup -> set up
 instruction -> instructions

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,422 @@
+3. Install dependencies for the Flutter demo app:
+```bash
+# run this command inside the flutter project directory `demo_app/`
+flutter pub add camera intl;
````
Is the semicolon needed here?

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,422 @@
+
+
+### 3. Setup askui
+1. Setup askui by following [these steps](https://docs.askui.com/docs/general/Getting%20Started/getting-started).
```
I recommend to replace _these steps_ with _the Getting Started Guide_

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,422 @@
+
+### 3. Setup askui
+1. Setup askui by following [these steps](https://docs.askui.com/docs/general/Getting%20Started/getting-started).
+2. We need to run the uiController directly with an extra argument to specify the runtime mode, as the current version of askui(ver. 0.3.2) doesn't provide the api for running it with the runtime argument yet. <br><br>From within your npm project path, go to the directory that contains the askui-ui-controller binary, and run `./askui-ui-controller -r android`
```
More nitpicking:

uiController -> UiController (Derived from the class in TypeScript)

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,422 @@
+npx jest test/my-first-askui-test-suite.test.ts --config ./test/jest.config.ts
+```
+
+3. If you got them both(emulator and uiController) running, then we are ready to go for the UI automation.<br><br>If you are working with the test code from our [official docs](https://docs.askui.com/docs/general/Getting%20Started/writing-your-first-test/), then you need to deactivate few lines of the code in `test/helper/jest.setup.ts` that is running the uiController, because we are already run ...
````
both(emulator and uiController) -> both (emulator and UiController)

HTML is possible but not desirable in Markdown: Adding a blank line should give an extra paragraph.

few lines -> a few lines

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,422 @@
+```
+
+--------
+## Breaking down the askui test code 
````
## Breaking Down the Askui Test Code

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,422 @@
+1. **Try to annotate** : Use `await aui.annotateInteractively();` or `await aui.annotate();` in order to see how askui is understanding the visible elements on your screen. By using `await aui.annotate()`, the result of the annotation will be saved in `report/` as an HTML file.
+2. **Be aware of the screen size of your device** : askui understands your application based on the screen shown and captured. Therefore, in some occasions, you may want to know your screen size in order to e.g. properly scroll or swipe within your application. You may need to change the numbers for the `input swipe` command within the provided test code, so that it suits the screen size of your  ...
+
+    - *tip: If you are using a device with a bigger screen e.g. Tablet, then the screen of your test device(real android device or emulator) might be big enough to see the whole page without scrolling. Then, you can remove/comment out the code lines that are scrolling the page.*
```
device(real... -> device (real...

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,422 @@
+        await aui.type('askui').exec();
+        await aui.click().text().withText('Enter your email').exec();
+        await aui.type('askui@askui.com').exec();
+        /*  Scroll down the page
```
Perfect comment to explain this!

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,422 @@
+        // Click and type the address
+        await aui.click().text().withText('Enter your address').exec();
+        await aui.type('Haid-und-Neu-Straße 18').exec();
+        // Pressing enter is the equivelant to pressing the return button on the on-screen-keyboard
```
equivelant -> equivalent

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,422 @@
-# flutter-example-automation
-Basic Flutter app with askui automation examples
+# Askui automation example with Flutter
```
# Askui Automation Example With Flutter

JohannesDienst-askui commented on README.md:
```diff
@@ -1,2 +1,422 @@
+- Type automation
+- Swipe automation
+
+#### Live demo in action (playback speed x3)
```
#### Live Demo in Action (playback speed x3)

JohannesDienst-askui reviewed (changes requested):
Unfortunately I found some spelling mistakes and Heading capitalization issues.

Overall great though!

HaramChoi-askui committed:
fixed: README.md

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,423 @@
+npx jest test/my-first-askui-test-suite.test.ts --config ./test/jest.config.ts
+```
+
+3. If you got them both(emulator and UiController) running, then we are ready to go for the UI automation.
````
Please add a space between both and (

JohannesDienst-askui commented on README.md:
````diff
@@ -1,2 +1,423 @@
+```
+
+3. If you got them both(emulator and UiController) running, then we are ready to go for the UI automation.
+If you are working with the test code from our [official docs](https://docs.askui.com/docs/general/Getting%20Started/writing-your-first-test/), then you need to deactivate a few lines of the code in `test/helper/jest.setup.ts` that is running the uiController, because we are already running it manually in the previous step.
````
uiController -> UiController

HaramChoi-askui committed:
fixed: README.md, demo-automation.ts, images/

After the release of ver. 0.4, description for the landscape-mode-only has been deprecated.
The description for it has been removed.
The demo images are replaced to portrait mode images.

HaramChoi-askui committed:
fixed: Images now smaller

HaramChoi-askui committed:
fixed: README.md
